modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/workflow/WorkflowExecutorFactory.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    /**
+     * Load configured class to the class loader and
+     * create instance of the particular class
+     */
+    public void loadWorkflowConfiguration() throws AppFactoryException {
+        if (log.isDebugEnabled()) {
+            String message = "Load configured class to class loader and create instance of that particular class "
+                    + "dynamically";
+            log.debug(message);
+        }
+
+        workflowExecutorMap = new ConcurrentHashMap<WorkflowConstant.WorkflowType, WorkflowExecutor>();
+        String executorClass;
+        WorkflowExecutor workflowExecutor;
+        try {
+            executorClass = AppFactoryUtil.getAppfactoryConfiguration()
+                    .getFirstProperty(WorkflowConstant.TENANT_CREATION);
+        } catch (AppFactoryException e) {
+            String message =
+                    "Unable to read executor class from the appfactory.xml configuration, the workflow type : "
+                            + WorkflowConstant.WorkflowType.TENANT_CREATION;
+            throw new AppFactoryException(message, e);
+        }
+
+        if (StringUtils.isBlank(executorClass)) {
+            executorClass = WorkflowConstant.TENANT_CREATION_DEFAULT_WORKFLOW_CLASS;
+            String message = "The tenant creation executor class is not define in appfactory.xml because tenant "
+                    + "creation default workflow is " + "executing. The tenant creation default workflow class is : "
+                    + executorClass;
+            log.warn(message);
+        }
+
+        try {
+            Class tenantCreationExecutorClass = WorkflowExecutorFactory.class.getClassLoader().loadClass(executorClass);
+            workflowExecutor = (WorkflowExecutor) tenantCreationExecutorClass.newInstance();
+            workflowExecutorMap.put(WorkflowConstant.WorkflowType.TENANT_CREATION, workflowExecutor);
+        } catch (ClassNotFoundException e) {
+            String message = "Unable to find configured class : " + executorClass + " in the class loader, when "
+                    + "initializing the executor class for workflow type : "
+                    + WorkflowConstant.WorkflowType.TENANT_CREATION;
+            throw new AppFactoryException(message, e);
+        } catch (InstantiationException e) {
+            String message = "Unable to initiate object of the class : " + executorClass + "when initializing the "
+                    + "object of the executor class for workflow type : "
+                    + WorkflowConstant.WorkflowType.TENANT_CREATION;
+            throw new AppFactoryException(message, e);
+        } catch (IllegalAccessException e) {
+            String message = "Unable to access the class" + executorClass + " for create new instance when accessing "
+                    + "the executor class for workflow type : " + WorkflowConstant.WorkflowType.TENANT_CREATION;
+            throw new AppFactoryException(message, e);
+        }
+    }
+
```

Comment:
No need to log the error here. It is logged in https://github.com/wso2/product-af/pull/664/files#diff-032f69c60db0d971fd78c93eeb08da74R46